Foundation/GTMTimeUtils.m

```diff
@@ -0,0 +1,70 @@
+//
+//  GTMTimeUtils.m
+//
+//  Copyright 2018 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+//  use this file except in compliance with the License.  You may obtain a copy
+//  of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+//  License for the specific language governing permissions and limitations under
+//  the License.
+//
+
+#import "GTMTimeUtils.h"
+
+#include <sys/sysctl.h>
+
+NSTimeInterval GTMTimeValToNSTimeInterval(struct timeval time) {
+  return time.tv_sec + (time.tv_usec / (double)USEC_PER_SEC);
+}
+
+struct timeval GTMBootTimeRelativeTo1970(void) {
+  struct timeval bootTime = { 0, 0 };
+  int mib[2] = { CTL_KERN, KERN_BOOTTIME };
+  size_t bootTimeSize = sizeof(bootTime);
+  if (sysctl(mib, 2, &bootTime, &bootTimeSize, NULL, 0) != 0) {
+    NSCAssert(errno == 0, @"sysctl error - %d", errno);
```

Comment:
These will be at the mercy of the developers setting to compile out in release?  Do you want to use our DevAssert to ensure they default to DEBUG only?

Reply:
Done